File: Unigram/Unigram.Tasks/TelegramPaymentProxy.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Data.Json;
using Windows.Foundation.Metadata;

namespace Unigram.Tasks
{
    public delegate void TelegramPaymentProxyDelegate(string title, string credentials);

    [AllowForWeb]
    public sealed class TelegramPaymentProxy
    {
        private TelegramPaymentProxyDelegate _callback;

        public TelegramPaymentProxy(TelegramPaymentProxyDelegate callback)
        {
            _callback = callback;
        }

        public void PostEvent(string eventName, string eventData)
        {
            if (eventName.Equals("payment_form_submit"))
            {
                try
                {
                    var json = JsonObject.Parse(eventData);
                    var response = json.GetNamedValue("credentials");
                    var title = json.GetNamedString("title", string.Empty);

                    _callback.Invoke(title, response.Stringify());
                }
                catch
                {
                    _callback.Invoke(null, eventData);
                }
            }
        }
    }
}

File: Unigram/Unigram.Native/TelegramPaymentProxy.cpp
#include "pch.h"
#include "TelegramPaymentProxy.h"

#include <winrt/Windows.Data.Json.h>

using namespace winrt::Windows::Data::Json;

namespace winrt::Unigram::Native::implementation
{
	TelegramPaymentProxy::TelegramPaymentProxy(TelegramPaymentProxyDelegate delegate)
	{
		_delegate = delegate;
	}

	void TelegramPaymentProxy::PostEvent(hstring eventName, hstring eventData)
	{
		if (eventName == L"payment_form_submit")
		{
			try
			{
				auto json = JsonObject::Parse(eventData);
				auto response = json.GetNamedValue(L"credentials");
				auto title = json.GetNamedString(L"title", L"");

				_delegate(title, response.Stringify());
			}
			catch (winrt::hresult_error const& ex)
			{
				_delegate(L"", eventData);
			}
		}
	}
} // namespace winrt::Unigram::Native::implementation